src/parallel-transform.js

```diff
@@ -1,194 +0,0 @@
-/* eslint-disable no-underscore-dangle */
-import stream from 'stream';
-import cyclist from 'cyclist';
-
-const _maxParallel = new WeakMap(),
-  _destroyed = new WeakMap(),
-  _flushed = new WeakMap(),
-  _buffer = new WeakMap(),
-  _top = new WeakMap(),
-  _bottom = new WeakMap(),
-  _ondrain = new WeakMap();
-
-export default class ParallelTransform extends stream.Transform {
-  /**
-   * ParallelTransform instance
-   * All child classes must implement the `_parallelTransform` function.
-   * Child classes should not implement the `_transform` and `_flush` functions.
-   *
-   * @param {Object} options Options which will be passed to the
-   *                         `stream.Transform` constructor
-   **/
-  constructor(options = {}) {
-    const maxParallel = options.maxParallel || 5,
-      defaultOptions = {
-        maxParallel,
-        highWaterMark: Math.max(maxParallel, 16)
-      };
-
-    super(Object.assign({}, defaultOptions, options));
-
-    // set default properties
-    _maxParallel.set(this, maxParallel);
-    _destroyed.set(this, false);
-    _flushed.set(this, false);
-    _buffer.set(this, cyclist(maxParallel));
-    _top.set(this, 0);
-    _bottom.set(this, 0);
-    _ondrain.set(this, null);
-  }
-
-  /**
-   * Helper function for easily creating ParallelTransform streams
-   *
-   * @param {Function} transform The stream's _transform function
-   * @param {Function} flush The stream's _flush function
-   * @param {Object} defaultOptions Default options for the class constructor
-   * @returns {Class} A ParallelTransform class
-   **/
-  static create(transform, flush = done => done(), defaultOptions = {}) {
-    class Transform extends ParallelTransform {
-      constructor(options) {
-        super(Object.assign({}, defaultOptions, options));
-      }
-
-      _parallelTransform = transform;
-      _parallelFlush = flush;
-    }
-
-    return Transform;
-  }
-
-  /**
-   * Destroys the stream
-   * The results of all pending transformations will be discarded
-   **/
-  destroy() {
-    if (_destroyed.get(this)) {
-      return;
-    }
-
-    _destroyed.set(this, true);
-    this.emit('close');
-  }
-
-  /**
-   * Parallises calls to this._transformFunction
-   * @param {?}        chunk The chunk of data to be transformed
-   * @param {string}   encoding Encoding, if it `chunk` is a string
-   * @param {Function} done Callback to be called when finished
-   **/
-  _transform(chunk, encoding, done) {
-    const pos = _top.get(this);
-    _top.set(this, pos + 1);
-
-    this._parallelTransform(chunk, encoding, (err, data) => {
-      if (_destroyed.get(this)) {
-        return;
-      }
-
-      // abort on error
-      if (err) {
-        this.emit('error', err);
```

Comment:
Isn’t the convention that the error is the first var? See how you use the `_paralllelTransform`.